omod-common/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/MainResourceController.java

```diff
@@ -123,9 +123,17 @@ public Object update(@PathVariable("resource") String resource, @PathVariable("u
 	        throws ResponseException {
 		baseUriSetup.setup(request);
 		RequestContext context = RestUtil.getRequestContext(request, response);
-		Updatable res = (Updatable) restService.getResourceByName(buildResourceName(resource));
-		Object updated = res.update(uuid, post, context);
-		return RestUtil.updated(response, updated);
+		
+		if (post.get("deleted") != null && "false".equals(post.get("deleted")) && post.size() == 1) {
+			Deletable res = (Deletable) restService.getResourceByName(buildResourceName(resource));
+			Object undeletedRes = res.undelete(uuid, context);
+			return RestUtil.updated(response, undeletedRes);
+		}
+		else {
+			Updatable res = (Updatable) restService.getResourceByName(buildResourceName(resource));
```

Comment:
What happens if you have the deleted property together with others?

Reply:
Also what happens when the property deleted has a value of true?

Reply:
An Exception similar to all other invalid properties is thrown, i.e conversion Exception.

Reply:
Do you have a test to confirm so?

Reply:
This test should be part of work done earlier since this is not the first time this possibility exists.